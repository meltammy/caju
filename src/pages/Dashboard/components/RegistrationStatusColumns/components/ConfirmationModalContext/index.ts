import { createContext } from "react";
import { RegistrationStatus } from "~/types";

export type ConfirmationModalProps =
  | {
      changeStatusProps?: {
        id: string;
        status: RegistrationStatus;
      };
      deleteRegistrationProps?: {
        id: string;
      };
    }
  | undefined;

export interface ConfirmationModalContextType {
  confirmationModalProps: ConfirmationModalProps;
  setConfirmationModalProps: (value: ConfirmationModalProps) => void;
}

export const ConfirmationModalContext = createContext<
  ConfirmationModalContextType | undefined
>(undefined);
