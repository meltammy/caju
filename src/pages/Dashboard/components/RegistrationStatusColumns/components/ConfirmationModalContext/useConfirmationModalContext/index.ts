import { useContext } from "react";
import { ConfirmationModalContext, ConfirmationModalContextType } from "..";

export function useConfirmationModalContext(): ConfirmationModalContextType {
  const context = useContext(ConfirmationModalContext);

  if (!context) {
    throw new Error(
      "useChangeRegistrationStatusContext must be used within a SearchRegistrationProvider"
    );
  }

  return context;
}
