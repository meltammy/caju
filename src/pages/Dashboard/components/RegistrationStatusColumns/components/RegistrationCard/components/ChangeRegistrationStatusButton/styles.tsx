import { HTMLAttributes } from "react";
import styled from "styled-components";
import { RegistrationStatus } from "~/types";

type Props = {
  status: RegistrationStatus;
} & HTMLAttributes<HTMLButtonElement>;

export const ButtonStatus = styled.button<Props>`
  font-size: 12px;
  outline: none;
  border-radius: 4px;
  border: none;
  padding: 4px 16px;
  font-size: 1rem;
  font-weight: 600;
  width: 100%;
  border-radius: 0;
  height: 3rem;
  text-transform: uppercase;
  color: ${({ theme, status }) => theme.registrationStatus[status].buttonColor};
  background: transparent;
  border-top: 1px ${({ theme }) => theme.colors.grey} solid;

  &:first-of-type {
    border-right: 1px ${({ theme }) => theme.colors.grey} solid;
  }
`;
