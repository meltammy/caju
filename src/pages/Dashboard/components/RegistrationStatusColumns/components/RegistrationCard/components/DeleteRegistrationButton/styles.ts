import styled from "styled-components";
import { IconButton } from "~/components/Buttons/IconButton";

export const StyledIconButton = styled(IconButton)`
  position: absolute;
  top: 1.3rem;
  right: 1rem;
`;
