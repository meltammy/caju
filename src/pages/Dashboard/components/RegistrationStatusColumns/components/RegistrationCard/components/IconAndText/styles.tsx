import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
  gap: 0.5rem;
  padding: 0 1rem;
  font-size: 1rem;

  svg {
    min-width: 1rem;
  }

  span {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &:first-of-type {
    margin-bottom: 0.5rem;
    padding: 1rem 3rem 0.5rem 1rem;
    border-bottom: 1px black solid;
    font-size: 2rem;

    span {
      margin: 0;
      white-space: break-spaces;
      font-size: 1.5rem;
      font-weight: 600;
      line-height: 2rem;
    }
  }
`;
