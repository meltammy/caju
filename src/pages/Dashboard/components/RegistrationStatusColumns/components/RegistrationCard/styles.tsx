import styled from "styled-components";

export const Card = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  gap: 4px;
  border: 1px solid ${({ theme }) => theme.colors.lightGrey};
  margin: 16px;
  border-radius: 8px;
  background-color: ${({ theme }) => theme.colors.white};
  overflow: hidden;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);

  h3,
  p {
    margin: 0;
  }
`;

export const IconAndText = styled.div`
  display: flex;
  align-items: center;
  gap: 8px;
  font-size: 1rem;
  padding: 0 1rem;

  &:first-of-type {
    font-size: 2rem;
    border-bottom: 1px black solid;
    padding: 1rem 3rem 0.5rem 1rem;
    margin-bottom: 0.5rem;

    h2 {
      margin: 0;
      font-size: 1.5rem;
      font-weight: 600;
      line-height: 1.5rem;
    }
  }
`;

export const Actions = styled.div`
  margin-top: 8px;
  display: flex;
`;
