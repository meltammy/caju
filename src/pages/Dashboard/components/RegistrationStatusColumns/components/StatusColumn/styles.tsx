import styled from "styled-components";
import { RegistrationStatus } from "~/types";

export const Column = styled.div<{ $status: RegistrationStatus }>`
  height: auto;
  background-color: ${({ $status: status, theme }) =>
    theme.registrationStatus[status].background};
  border-radius: 2rem;
  height: 100%;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    min-width: 18rem;

    &:nth-child(3) {
      margin-right: 1.5rem;
    }
  }
`;

export const TitleColumn = styled.h2<{ $status: any }>`
  color: ${({ $status: status, theme }) =>
    theme.registrationStatus[status].color};
  margin: 1.5rem;
`;

export const ColumContent = styled.div`
  height: 100%;
`;

export const NoResultsMessage = styled.h3`
  margin: auto;
  margin: 15rem 1rem;
  text-align: center;
  color: ${({ theme }) => theme.colors.lightGrey};
`;
