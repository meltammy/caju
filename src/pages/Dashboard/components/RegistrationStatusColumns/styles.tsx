import styled from "styled-components";
import { Spinner } from "~/components/Spinner";

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 1.5rem;
  justify-content: center;
  min-height: 68vh;
  margin-bottom: 1rem;

  @media screen and (max-width: 1200px) {
    justify-content: flex-start;
    overflow: auto;
    margin-left: 24px;
  }
`;

export const StyledSpinner = styled(Spinner)`
  grid-column: 2;
`;
