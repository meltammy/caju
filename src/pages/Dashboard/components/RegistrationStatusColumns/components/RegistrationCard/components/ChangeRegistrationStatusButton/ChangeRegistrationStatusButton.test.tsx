import "jest-styled-components";
import "@testing-library/jest-dom";

import { render, screen, fireEvent } from "@testing-library/react";
import { ChangeRegistrationStatusButton } from ".";
import { RegistrationStatus } from "~/types";
import { ThemeProvider } from "styled-components";
import { theme } from "~/../styles/theme";
import { ReactNode } from "react";
import { useConfirmationModalContext } from "~/pages/Dashboard/components/ConfirmationModals/ConfirmationModalContext/useConfirmationModalContext";
import { ConfirmationModalContextType } from "~/pages/Dashboard/components/ConfirmationModals/ConfirmationModalContext";

jest.mock(
  "~/pages/Dashboard/components/ConfirmationModals/ConfirmationModalContext/useConfirmationModalContext"
);

const renderWithThemeAndContext = (
  children: ReactNode,
  mockContextValue: Partial<ConfirmationModalContextType>
) => {
  (useConfirmationModalContext as jest.Mock).mockReturnValue(mockContextValue);
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

describe("ChangeRegistrationStatusButton", () => {
  const mockSetConfirmationModalProps = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders with the correct label based on status", () => {
    const mockContextValue = {
      setConfirmationModalProps: mockSetConfirmationModalProps,
    };
    const status = RegistrationStatus.Approved;

    renderWithThemeAndContext(
      <ChangeRegistrationStatusButton status={status} id="1" />,
      mockContextValue
    );

    expect(screen.getByText("Aprovar")).toBeInTheDocument();
  });

  it("calls setConfirmationModalProps with the correct parameters on click", () => {
    const mockContextValue = {
      setConfirmationModalProps: mockSetConfirmationModalProps,
    };
    const status = RegistrationStatus.Reproved;

    renderWithThemeAndContext(
      <ChangeRegistrationStatusButton status={status} id="2" />,
      mockContextValue
    );

    fireEvent.click(screen.getByRole("button"));

    expect(mockSetConfirmationModalProps).toHaveBeenCalledWith({
      changeStatusProps: { status, id: "2" },
    });
  });

  it("renders the button with the correct id", () => {
    const mockContextValue = {
      setConfirmationModalProps: mockSetConfirmationModalProps,
    };
    const status = RegistrationStatus.Review;

    renderWithThemeAndContext(
      <ChangeRegistrationStatusButton status={status} id="3" />,
      mockContextValue
    );

    const button = screen.getByRole("button");
    expect(button).toHaveAttribute("id", `button-change-status-to-${status}`);
  });

  it("matches snapshot", () => {
    const mockContextValue = {
      setConfirmationModalProps: mockSetConfirmationModalProps,
    };
    const status = RegistrationStatus.Approved;

    const { asFragment } = renderWithThemeAndContext(
      <ChangeRegistrationStatusButton status={status} id="1" />,
      mockContextValue
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
