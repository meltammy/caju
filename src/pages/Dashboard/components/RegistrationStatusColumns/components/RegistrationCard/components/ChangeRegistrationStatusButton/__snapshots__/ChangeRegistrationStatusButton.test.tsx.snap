// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeRegistrationStatusButton matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  height: 3rem;
  padding: 0.25rem 1rem;
  border-radius: 0.25rem;
  border: none;
  border-radius: 0;
  font-size: 0.75rem;
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  color: #0066ff;
  background: transparent;
  border-top: 1px #ccc solid;
}

.c0:first-of-type {
  border-right: 1px #ccc solid;
}

<button
    class="c0"
    id="button-change-status-to-APPROVED"
  >
    Aprovar
  </button>
</DocumentFragment>
`;
